src/Compilers/CSharp/Portable/Symbols/Source/ParameterHelpers.cs

```diff
@@ -337,11 +337,11 @@ internal static MethodSymbol FindContainingGenericMethod(Symbol symbol)
             return null;
         }
 
-        private static RefKind GetModifiers(SyntaxTokenList modifiers, out SyntaxToken argPassingKeyword, out SyntaxToken paramsKeyword, out SyntaxToken thisKeyword)
+        private static RefKind GetModifiers(SyntaxTokenList modifiers, out SyntaxToken refOrOutKeyword, out SyntaxToken paramsKeyword, out SyntaxToken thisKeyword)
```

Comment:
refOrOutKeyword   is only needed to get    "ref" or "out"  modifiers, which are not allowed in property/indexer parameters. We do not need to return "in"/"ref readonly" tokens since those are allowed everywhere.

We also need tokens for "this" and "params" presence of those cannot be figured from RefKind, and we also need the actual tokens to report errors.

---
In reply to: [103585079](https://github.com/dotnet/roslyn/pull/17469#discussion_r103585079) [](ancestors = 103585079)